### PR TITLE
Issue 18341 Unable to add content to pages using MySQ

### PR DESCRIPTION
**Pull request by freddyucv**
This query

https://github.com/dotCMS/core/compare/issue-18343-Unable-to-create-a-site-search-index-using-MySQL-as-database?expand=1#diff-27a76393f6a1bc82519eb4b4291b3851R82

it was failing in My SQL, the reason for this error is that MySQL doesn’t allow updates to a table when you are also using that same table in an inner select as your update criteria.

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/datagen/MultiTreeDataGen.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import com.dotmarketing.portlets.containers.model.Container;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+
+public class MultiTreeDataGen extends AbstractDataGen<MultiTree> {
+    private Container container;
+    private Contentlet contentlet;
+    private String instanceID;
+    private String personalization;
+    private int treeOrder;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/MultiTreeDataGen.java:15]: Codacy Issue found: [Avoid unused private fields such as 'treeOrder'.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5411384)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/MultiTreeDataGen.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.portlets.containers.model.Container;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+
+public class MultiTreeDataGen extends AbstractDataGen<MultiTree> {
+    private Container container;
+    private Contentlet contentlet;
+    private String instanceID;
+    private String personalization;
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/MultiTreeDataGen.java:14]: Codacy Issue found: [Avoid unused private fields such as 'personalization'.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5411384)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/MultiTreeDataGen.java (hunk 1 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/MultiTreeDataGen.java:15]: Codacy Issue found: [Perhaps 'treeOrder' could be replaced by a local variable.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5411384)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/MultiTreeDataGen.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        return multiTree;
+    }
+
+    @Override
+    public MultiTree persist(final MultiTree multiTree) {
+        try {
+            APILocator.getMultiTreeAPI().saveMultiTree(multiTree);
+            return multiTree;
+        } catch (DotDataException e) {
+            throw  new RuntimeException(e);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/MultiTreeDataGen.java:37]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5411384)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/MultiTreeDataGen.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    private Container container;
+    private Contentlet contentlet;
+    private String instanceID;
+    private String personalization;
+    private int treeOrder;
+    private HTMLPageAsset page;
+
+    @Override
+    public MultiTree next() {
+        MultiTree multiTree = new MultiTree();
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/MultiTreeDataGen.java:20]: Codacy Issue found: [Local variable 'multiTree' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5411384)

File: dotCMS/src/integration-test/java/com/dotcms/datagen/MultiTreeDataGen.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/datagen/MultiTreeDataGen.java:14]: Codacy Issue found: [Perhaps 'personalization' could be replaced by a local variable.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5411384)

**Review by erickgonzalez — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

**Pull request merged**